### PR TITLE
docs: add guide for replicating onchain state

### PR DESCRIPTION
**Pull request by holic**
https://mud-docs-git-holic-store-state-from-onchain-latticexyz.vercel.app/guides/replicating-onchain-state

### CODE REVIEW

File: docs/pages/guides/replicating-onchain-state.mdx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+## Store: MUD’s onchain storage
+
+First, it’s helpful to understand how MUD stores its data on chain. Like a native Solidity `mapping`, MUD stores values at keys. However, it handles both keys and values in a particular way to 1) give a bit more flexibility in data modeling and 2) squeeze out a bit more gas.
+
+- In MUD, keys are represented as a combination of table ID and key tuple (i.e. a composite key). Internally, the storage slot is derived from a hash of the table ID, key tuple, and “section” of data (static or dynamic).
+- All static (fixed length) fields are tightly packed into a single bytes value stored in a single storage slot. Two `uint32` fields and a `bool` field would be packed into 9 bytes (4 + 4 + 1).
+- For all dynamic (variable length) fields, we encode and store a “dynamic data layout” (also called “encoded lengths”) that represents the length of each dynamic field (i.e. number of items in an array, length of string or bytes). Each dynamic field is given its own storage slot that contains its value.
```
Comment (yonadaaa) [inline docs/pages/guides/replicating-onchain-state.mdx:11]: These bullet points are great (very glad to have an explanation of this stuff), but it's a lot to process on first glance with all the new terms, brackets, and quotation marks. 

A diagram or even glossary of terms could help!

File: docs/pages/guides/replicating-onchain-state.mdx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  const record = store.get(key) ?? { staticData: "0x", encodedLengths: "0x", dynamicData: "0x" };
+  store.set(key, {
+    staticData: record.staticData,
+    encodedLengths: log.args.encodedLengths,
+    dynamicData: bytesSplice(record.dynamicData, log.args.start, log.args.deleteCount, log.args.data),
+  });
+}
+```
+
+And that’s it! We’ve got our onchain state represented in TypeScript, reconstructed via MUD’s Store event logs.
```
Comment (yonadaaa) [inline docs/pages/guides/replicating-onchain-state.mdx:126]: Maybe show the `console.log` of what the state might look like? I find it helpful to have tangible examples

**Review by yonadaaa — COMMENTED** [review]
Fantastic explanation and so cool to look at the stack from first principles 👨‍🏫

File: docs/pages/guides/replicating-onchain-state.mdx (hunk 2 above)
Comment (holic) [reply docs/pages/guides/replicating-onchain-state.mdx:126]: it's all just hex, so I'm not sure how useful that would be 🙈 and I am hoping most readers can understand/imagine what a splice operation does?

Comment (holic) [reply docs/pages/guides/replicating-onchain-state.mdx]: I've reworded this and now link out to the other docs pages that provide more detail on the encoding.

File: docs/pages/guides/replicating-onchain-state.mdx (hunk 1 above)
Comment (holic) [reply docs/pages/guides/replicating-onchain-state.mdx]: I've reworded this and now link out to the other docs pages that provide more detail on the encoding.

Comment (yonadaaa) [reply docs/pages/guides/replicating-onchain-state.mdx]: Super clear and to the point now!! Sad you had to remove your text, maybe it can fit elsewhere

**Review by yonadaaa — APPROVED** [review]

**Pull request merged**